### PR TITLE
CFE-654: Feature/olm scaffolding

### PR DESCRIPTION
**Pull request by thejasn**
## Description
---
Migration of cert manager operator to use `operator-sdk` for OLM bundle generation and packaging. It changes how bundles are bootstrapped and tweaked. 

OLM scaffolding generated using,
```go
$> operator-sdk init --domain openshift.io --project-name cert-manager-operator --repo github.com/openshift/cert-manager-operator
$> operator-sdk create api --group operator.openshift.io --kind CertManager --namespaced=false --version v1alpha1
```

## Notable Changes
- Operator namespace changed from `openshift-cert-manager-operator` -> `cert-manager-operator`.
- Operand namespace changed from `openshift-cert-manager` -> `cert-manager`.
- `config/alphav1` API dropped for the operator.
- Bundle version changed to `0.0.1` (default).
- New `Dockerfile` introduced for CI.

## Change Log
- `apis` -> `api`
    - Renamed since `config/alphav1` API was removed.
- `bundle` : Regenerated using `operator-sdk`.
    - Operator CSV updated
    - Operator rbac regenerated using kubebuilder annotations
    - Operator install mode changed from `AllNamespaces` to `SingleNamespace` and `OwnedNamespace`.
- `cmd` : Deleted and `main.go` moved to project root.
- `config` : Bundle template generated using `operator-sdk`.
- `hack`
    - `hack/update-cert-manager-manifests.sh` : Update manifest destination as `config/crd/bases/`.
    - `hack/update-clientgen.sh` and `hack/update-deepcopy.sh` : Update `api` file path.
- `jsonnet` : Update script with the new namespaces.
- `pkg/*`
    - `pkg/cmd/operator/cmd.go` : Update operator namespace
    - `pkg/controller/certmanager_controller.go` : Introduced placeholder controller along with required kubebuilder annotations.
    - `pkg/operator` : Re-generate operator client and helper functions.

### CODE REVIEW

File: bundle/manifests/cert-manager-operator.clusterserviceversion.yaml
Comment (ayesha54) [inline bundle/manifests/cert-manager-operator.clusterserviceversion.yaml]: shall we customize the image name here? @thejasn

File: pkg/controller/deployment/cert_manager_controller_deployment.go
Comment (TrilokGeer) [inline pkg/controller/deployment/cert_manager_controller_deployment.go]: Is this updated in any of cr containing rbac configurations? Or is it to avoid clusterrole for the operator.openshift.io.openshift.io group?

File: bundle/manifests/cert-manager-operator.clusterserviceversion.yaml
Comment (TrilokGeer) [inline bundle/manifests/cert-manager-operator.clusterserviceversion.yaml:386]: What is the intent to have kube-rbac-proxy?

Comment (thejasn) [reply bundle/manifests/cert-manager-operator.clusterserviceversion.yaml:386]: This will come into picture when we want to expose operator metrics to console. As for the port change, the issue is that there is some conflict with 8443 with the operator also listening on the same port by default (https://github.com/openshift/library-go/blob/8ce03fe3cf2269c39e9fbd04473053a0925d0b1f/pkg/controller/controllercmd/cmd.go#L62). We can get around to fixing this post this OLM PR.

File: images/ci/Dockerfile
Comment (thejasn) [reply images/ci/Dockerfile]: Fixed

File: pkg/controller/deployment/cert_manager_controller_deployment.go
Comment (thejasn) [reply pkg/controller/deployment/cert_manager_controller_deployment.go]: Already addressed in `certmanager_controller.go`.

**Comment by thejasn** [comment]
/test e2e-operator
Unrelated AWS side error.

**Comment by thejasn** [comment]
/hold 
For QE approval

**Comment by xingxingxia** [comment]
Checked through PR's description for the changes and some files. Adding qe-approved for requirement from https://github.com/openshift/release/pull/34826
/label qe-approved

**Comment by thejasn** [comment]
Docs label: /assign @bergerhoffer  
As far as docs goes, this PR sets the foundation for operator-sdk based packaging only (will be followed up with additional PRs).

PX label: /assign @davemulford

**Comment by thejasn** [comment]
/cc @bergerhoffer @davemulford 
As far as docs goes, this PR sets the foundation for operator-sdk based packaging only (will be followed up with additional PRs).

**Comment by bergerhoffer** [comment]
@thejasn docs doesn't normally have to approve eng PRs. If there is any user-facing display text that you'd like me to review before it merges in, I'd be happy to do that.

Or feel free to tag me as a heads up as things come in if they are specific new functionality that I might need to know for documentation purposes. Thanks!

**Comment by davemulford** [comment]
I saw the epic in Jira and provided comments there.
/label px-approved

**Review by geliu2016 — APPROVED** [review]
/label qe-approved

File: bundle/manifests/cert-manager-operator.clusterserviceversion.yaml
Comment (alebedev87) [inline bundle/manifests/cert-manager-operator.clusterserviceversion.yaml:79]: That is more verbs than used to be. Why?

**Review by alebedev87 — COMMENTED** [review]
Sorry for the last review.

As we discussed the main points are:
- API doesn't seem to be mature enough:
       - it allows too many configuration permutations some of which can be dangerous/not recommended/not tested
       - it doesn't have the vision of how CertManager is supposed to be used (not opinionated), the operator is just `oc apply`, the operational logic is not quite present
- Since the namespaces changed: a migration guide needs to be provided in this repository and in the OpenShift docs

The rest is just small remarks after a brief look through.

**Pull request merged**
